tools/release_packages

```diff
@@ -0,0 +1,372 @@
+    )
+    print(crayons.white(message, bold=True))
+
+    setup_dir = os.path.dirname(setup_file)
+    with cd(setup_dir):
+        setuptools.sandbox.run_setup(setup_file, setup_args)
+
+    message = '{dist_type} for {package_name} generated under {dir}.\n'.format(
+        dist_type=dist_type,
+        package_name=package_name,
+        dir=setup_dir,
+    )
+    print(crayons.green(message, bold=True))
+
+def generate_sdist(setup_file, package_name):
+    _generate_dist('sdist', setup_file, package_name, ['sdist'])
+
+def generate_bdist_wheel_universal(setup_file, package_name):
+    _generate_dist('bdist_wheel', setup_file, package_name,
+                   ['bdist_wheel', '--universal'])
+
+def twine_upload(dist_dirs):
+    message = 'Uploading distributions under the following directories:'
+    print(crayons.green(message, bold=True))
+    for dist_dir in dist_dirs:
+        print(crayons.yellow(dist_dir))
+    twine.commands.upload.main(dist_dirs)
+
+def cleanup(package_dir):
+    build_dir = os.path.join(package_dir, 'build')
+    temp_dir = os.path.join(package_dir, 'temp')
+    dist_dir = os.path.join(package_dir, 'dist')
+    egg_info = os.path.join(
+        package_dir,
+        '{}.egg-info'.format(os.path.basename(package_dir))
+    )
+    version_symlink = os.path.join(package_dir, 'version.py')
+
+    def _rm_if_it_exists(directory):
+        if os.path.isdir(directory):
+            print(crayons.green('Removing {}/*.'.format(directory), bold=True))
+            shutil.rmtree(directory)
+
+    map(_rm_if_it_exists, [build_dir, temp_dir, dist_dir, egg_info])
+
+    if os.path.islink(version_symlink):
+        print(crayons.green('Removing {}.'.format(version_symlink), bold=True))
+        os.remove(version_symlink)
+
+def increment_version(init_file, new_version):
+    fh, temp_abs_path = tempfile.mkstemp()
+    with os.fdopen(fh, 'w') as new_file, open(init_file) as old_file:
+        for line in old_file:
+            if line.startswith('__version__'):
+                prev = line.split('=')[-1].strip().replace('"', '')
+                new_file.write(line.replace(prev, new_version))
+            else:
+                new_file.write(line)
+
+    os.remove(init_file)
+    shutil.move(temp_abs_path, init_file)
+
+    message = 'Incremented __version__ in {fp} to {version}.'.format(
+        fp=init_file, version=new_version)
+    print(crayons.white(message, bold=True))
+
+def push_release_tag(version, upstream_or_origin):
+    print(crayons.yellow('Pushing release tag {}...'.format(version), bold=True))
+    git.tag(version)
+    git.push(upstream_or_origin, version)
+
+def commit_and_push_version_changes(version, init_files, upstream_or_origin):
+    message = 'Committing version number changes...{}'.format(version)
+    print(crayons.yellow(message, bold=True))
+
+    if upstream_or_origin == 'origin':
+        branch = 'release-{}'.format(version)
+        git.checkout('-b', branch)
+    else:
+        branch = 'master'
```

Comment:
Hm, I'm a little wary of this possibility, it seems to be potentially hazardous. Even though it makes the release process even shorter, would it be safer to make every release through PRs (one can merge them oneself, if needed). I see the danger in people with writing permission trying out this script.

Reply:
Well, I think it doesn't matter for the following reasons:
* Anyone without PyPA credentials shouldn't be using this script.
* For changes that can be pushed to the main repo, I trust anyone who has commit access to that repo to exercise their judgement and use `--push origin` and *not* `--push upstream`. If Tim or Steve or Greg want to push to master, they should be free to do so.
* For people who have commit access to this repo, I also trust them to not make a release without consulting me or senior developers like Tim (unless I am mistaken, I think Tim and I are the only ones with PyPA credentials for making a release).
* Even if someone were to push a commit to this repo changing the version numbers, that wouldn't necessarily break anything anywhere. The people installing the package from PyPA would get what they are supposed to get. And since the main repo installs off of a tag or commit, it would get what it wants regardless.

To sum up, I think this is okay to do. If you'd like to discuss this further, let me know! :)

Reply:
Okay, agreed.